presentation/charge_transaction.go

```diff
@@ -31,38 +31,68 @@ func (p *Presenter) ChargeTransaction(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	paymentType, err := parseValidPaymentMethod(requestBody)
+	if err != nil {
+		responseBody, e := json.Marshal(schema.Error{
+			StatusCode:    http.StatusBadRequest,
+			StatusMessage: err.Error(),
+		})
+		if e != nil {
+			w.WriteHeader(http.StatusInternalServerError)
+			return
+		}
+
+		w.Header().Set("Content-Type", "application/json")
+		w.WriteHeader(http.StatusBadRequest)
+		w.Write(responseBody)
+		return
+	}
+
+	var callbackURL string
+	if paymentType == primitive.PaymentTypeEMoneyGopay && requestBody.Gopay.EnableCallback {
+		callbackURL = requestBody.Gopay.CallbackURL
+	} else if paymentType == primitive.PaymentTypeEMoneyShopeePay {
+		callbackURL = requestBody.ShopeePay.CallbackURL
+	}
+
 	// Convert to common business schema
 	chargeRequest := business.ChargeRequest{
-		PaymentType:         paymentTypeMap[requestBody.PaymentType],
-		OrderId:             requestBody.Transaction.OrderId,
-		TransactionAmount:   requestBody.Transaction.Amount,
-		TransactionCurrency: currencyMap[requestBody.Transaction.Currency],
-
+		PaymentType:         paymentType,
+		OrderId:             requestBody.TransactionDetails.OrderId,
+		TransactionAmount:   requestBody.TransactionDetails.GrossAmount,
+		TransactionCurrency: currencyMap[requestBody.TransactionDetails.Currency],
 		Customer: business.CustomerInformation{
-			FirstName:   requestBody.Customer.FirstName,
-			LastName:    requestBody.Customer.LastName,
-			Email:       requestBody.Customer.Email,
-			PhoneNumber: requestBody.Customer.PhoneNumber,
+			FirstName:   requestBody.CustomerDetails.FirstName,
+			LastName:    requestBody.CustomerDetails.LastName,
+			Email:       requestBody.CustomerDetails.Email,
+			PhoneNumber: requestBody.CustomerDetails.PhoneNumber,
```

Comment:
ini single kan ya? harusnya singular ga? misal `requestBody.CustomerDetail.FirstName` 

```suggestion
			FirstName:   requestBody.CustomerDetail.FirstName,
			LastName:    requestBody.CustomerDetail.LastName,
			Email:       requestBody.CustomerDetail.Email,
			PhoneNumber: requestBody.CustomerDetail.PhoneNumber,
```


soalnya kuliat disini single customer aja kan? https://github.com/TokoBapak/mock-payment-provider/pull/17/files#

Reply:
nggak mau. ini kan targetnya mau 1:1 replica sama API nya midtrans. di midtrans dipakenya CustomerDetails bukan CustomerDetail.

kalo dibikin sesuai kaidah yang wajar, nggak jadi 1:1 replica.